tests/test_context.py

```diff
@@ -139,6 +140,33 @@ class TrackMe:
         del tracked
         self.assertIsNone(ref())
 
+    @unittest.skipUnless(PY37, 'requires Python 3.7')
+    def test_create_server_protocol_factory_context(self):
+        import contextvars
+        cvar = contextvars.ContextVar('cvar', default='outer')
+
+        def factory():
+            self.assertEqual(cvar.get(), 'inner')
+            return asyncio.Protocol()
+
+        async def test():
+            cvar.set('inner')
+            port = tb.find_free_port()
+            srv = await self.loop.create_server(factory, '127.0.0.1', port)
+
+            s = socket.socket(socket.AF_INET)
+            with s:
+                s.setblocking(False)
+                await self.loop.sock_connect(s, ('127.0.0.1', port))
+
+            srv.close()
+            await srv.wait_closed()
+
+            # Let all transports shutdown.
+            await asyncio.sleep(0.1)
```

Comment:
Using `sleep` is generally not recommended for new tests in uvloop - this test is actually unstable, I could get a false OK if the `srv.close()` got called before the connection is fully established.

A proper fix is to use a `Future`, set in `factory()` and awaited in `test()` - we could even pass over the assertion error in this manner. With this fix, the `ResourceWarning` is also gone.

Reply:
Ok, fixed